libs/common/src/main/java/org/opensearch/common/Booleans.java

```diff
@@ -45,30 +45,72 @@ private Booleans() {
     /**
      * Parses a char[] representation of a boolean value to <code>boolean</code>.
      *
-     * @return <code>true</code> iff the sequence of chars is "true", <code>false</code> iff the sequence of chars is "false" or the
-     * provided default value iff either text is <code>null</code> or length == 0.
+     * @return <code>true</code> iff the sequence of chars is "true", <code>false</code> iff the sequence of
+     * chars is "false" or the provided default value iff either text is <code>null</code> or length == 0.
      * @throws IllegalArgumentException if the string cannot be parsed to boolean.
      */
     public static boolean parseBoolean(char[] text, int offset, int length, boolean defaultValue) {
```

Comment:
Thanks for including the benchmarks - how did you run those tests?

What would it take to include them in this change so they can be reproduced - or more important repeated for on other optimization?

Reply:
The reason I didn't include the benchmarking code in this change is, given I'm replacing the baseline functionality, there'll be nothing to compare against.

Here's the gist: https://gist.github.com/ketanv3/c474156cad09fd208c2f11a7f9b2d20a
Run with `./gradlew -p benchmarks run --args 'BooleansBenchmark -prof gc'`

Let me know if you feel it needs to be included in this PR.

Reply:
Nice - thanks for letting me know.  Lets leave it out, but if anyone feels otherwise they can pull from the gist - much appreciated!